subsys/bluetooth/common/log.c

```diff
@@ -40,18 +47,32 @@ const char *bt_hex_real(const void *buf, size_t len)
 
 const char *bt_addr_str_real(const bt_addr_t *addr)
 {
+#ifdef CONFIG_LOG_IMMEDIATE
```

Comment:
Is this a problem only for LOG_IMMEDIATE? There could be valid reasons to have multiple calls to these functions i.e. generate a string for settings, etc.

Reply:
Yes, it is only a problem for LOG_IMMEDIATE, otherwise log_strdup will create a copy.

Reply:
You are saying log_strdup don't work when LOG_IMMEDIATE is set? Isn't it better to fix that, otherwise we have to duplicate this sort of buffering everywhere.

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/logging/log_core.c#L846
If not log immediate the same pointer is returned.

I think it is only a problem for these bt_*_str_real functions though, so I don't think we have to duplicate this sort of buffering anywhere else. This is how these functions was implement before the logging and log_strdup was introduced.

Reply:
I wonder if the way this buffer is used in bluetooth is not causing issues. Considering example:
```
LOG_INF("%s connected to %s",
         log_strdup(bt_addr_str_real(local_addr)),
         log_strdup(bt_addr_str_real(remote_addr)));
```
`LOG_IMMEDIATE` and proposed `LOG_MINIMAL` are turning log messages into simple, synchronous printf like calls and this approach is not ready for that.

Reply:
Ok, I'll accept that explanation and update the host to not use log_strdup in this way anymore.